File: attribouter/src/main/java/me/jfenn/attribouter/wedges/TextWedge.java
package me.jfenn.attribouter.wedges;

import android.content.Context;
import android.content.res.XmlResourceParser;
import android.os.Build;
import android.text.Html;
import android.text.method.LinkMovementMethod;
import android.view.View;
import android.widget.TextView;

import org.xmlpull.v1.XmlPullParserException;

import me.jfenn.attribouter.R;
import me.jfenn.attribouter.utils.ResourceUtils;

public class TextWedge extends Wedge {

    private String text;
    private boolean isCentered;

    public TextWedge(XmlResourceParser parser) throws XmlPullParserException {
        this(parser.getAttributeValue(null, "text"), parser.getAttributeBooleanValue(null, "centered", false));
    }

    public TextWedge(String text, boolean isCentered) {
        super(R.layout.item_attribouter_text);
        this.text = text;
        this.isCentered = isCentered;
    }

    @Override
    public ViewHolder getViewHolder(View v) {
        return new ViewHolder(v);
    }

    @Override
    public void bind(Context context, ViewHolder viewHolder) {
        TextView textView = (TextView) viewHolder.itemView;
        textView.setMovementMethod(new LinkMovementMethod());
        String text = ResourceUtils.getString(context, this.text);
        textView.setText(Build.VERSION.SDK_INT >= Build.VERSION_CODES.N ? Html.fromHtml(text, 0) : Html.fromHtml(text));
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1)
            textView.setTextAlignment(isCentered ? View.TEXT_ALIGNMENT_CENTER : View.TEXT_ALIGNMENT_GRAVITY);
    }

}

File: attribouter/src/main/java/me/jfenn/attribouter/wedges/TextWedge.kt
package me.jfenn.attribouter.wedges

import android.content.Context
import android.os.Build
import android.text.Html
import android.text.method.LinkMovementMethod
import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.google.android.flexbox.FlexDirection
import com.google.android.flexbox.FlexboxLayoutManager
import com.google.android.flexbox.JustifyContent
import me.jfenn.attribouter.R
import me.jfenn.attribouter.adapters.WedgeAdapter
import me.jfenn.attribouter.utils.ResourceUtils

open class TextWedge : Wedge<TextWedge.ViewHolder>(R.layout.attribouter_item_text) {

    private val text: String? by attr("text")
    private val isCentered: Boolean by attr("centered", false)

    override fun getViewHolder(v: View): ViewHolder {
        return ViewHolder(v)
    }

    override fun bind(context: Context, viewHolder: ViewHolder) {
        viewHolder.textView?.apply {
            movementMethod = LinkMovementMethod()

            val string = ResourceUtils.getString(context, this@TextWedge.text)
            text = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) Html.fromHtml(string, 0) else Html.fromHtml(string)

            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1)
                textAlignment = if (isCentered) View.TEXT_ALIGNMENT_CENTER else View.TEXT_ALIGNMENT_GRAVITY
        }

        val links = getTypedChildren<LinkWedge>().filter { !it.isHidden }.sorted()
        viewHolder.links?.apply {
            if (links.isNotEmpty()) {
                adapter = WedgeAdapter(links)
                layoutManager = FlexboxLayoutManager(context).apply {
                    flexDirection = FlexDirection.ROW
                    justifyContent = if (isCentered) JustifyContent.CENTER else JustifyContent.FLEX_START
                }

                visibility = View.VISIBLE
            } else visibility = View.GONE
        }
    }

    open class ViewHolder(v: View) : Wedge.ViewHolder(v) {
        var textView: TextView? = v.findViewById(R.id.text)
        var links: RecyclerView? = v.findViewById(R.id.links)
    }

}
